packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2959,6 +2961,101 @@ describe("Runtime", () => {
 					"SignalLatency telemetry should log absolute lost signal count for each batch of 100 signals and SignalOutOfOrder event",
 				);
 			});
+			it("ignores non-self-targeted signal in signalLatency telemetry", () => {
+				// Send 1st signal and process it to prime the system
+				sendSignals(1);
+				processSubmittedSignals(1);
+
+				// Rapid send 100+ signals (one targeted)
+				sendSignals(50);
+				containerRuntime.submitSignal(
+					"TargetedSignalType",
+					"TargetedSignalContent",
+					"mockTargetClient",
+				);
+				sendSignals(50);
+
+				// Process all signals
+				processSubmittedSignals(100);
+
+				// Check for logged SignalLatency event
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 100,
+							signalsLost: 0,
+							outOfOrderSignals: 0,
+						},
+					],
+					"SignalLatency telemetry should log correct amount of sent and lost signals",
+				);
+			});
+			it("includes self targeted signal in signalLatency telemetry", () => {
+				// Send 1st signal and process it to prime the system
+				sendSignals(1);
+				processSubmittedSignals(1);
+
+				// Rapid send 100+ signals (one self-targeted)
+				sendSignals(50);
+				containerRuntime.submitSignal(
+					"TargetedSignalType",
+					"TargetedSignalContent",
+					containerRuntime.clientId,
+				);
+				sendSignals(50);
+
+				// Process all signals
+				processSubmittedSignals(100);
+
+				// Check for logged SignalLatency event
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsSent: 100,
+							signalsLost: 0,
+							outOfOrderSignals: 0,
+						},
+					],
+					"SignalLatency telemetry should log correct amount of sent and lost signals",
+				);
+			});
+			it("can detect dropped signal while ignoring non-self targeted signal in signalLatency telemetry", () => {
+				// Send 1st signal and process it to prime the system
+				sendSignals(1);
+				processSubmittedSignals(1);
+
+				// Rapid send 100 signals (one targeted) and drop 10
```

Comment:
Doesn't feel like rapid send when 40 are sent and come back. (Rapid comment feels stale in prior cases too. I don't know what it is trying to convey.)

Reply:
Separated the sending from the processing for actual rapid fire signal simulation.